TezosKit/TezosNode/Services/RPCResponseHandler.swift

```diff
@@ -32,12 +32,23 @@ public class RPCResponseHandler {
       return .failure(rpcError)
     }
 
-    // Ensure that data came back.
-    guard let data = data,
-          let parsedData = parse(data, with: responseAdapterClass) else {
-      let tezosKitError = TezosKitError(kind: .unexpectedResponse, underlyingError: nil)
-      return .failure(tezosKitError)
-    }
+    // Check for data
+	guard let data = data, let stringData = String(data: data, encoding: .utf8) else {
+	  let tezosKitError = TezosKitError(kind: .unexpectedResponse, underlyingError: nil)
+	  return .failure(tezosKitError)
+	}
+
+	// Check for \"status\":\"backtracked\" in the response, indicating a failed transaction being rolledback
+	if stringData.contains("\"status\":\"backtracked\"") {
```

Comment:
Musing: Generally we can't count on this being true in all backwards compatible protocols (which is annoying). Happy to leave this here since we shouldn't really care about Athens / Babylon anymore, but eventually we may need to plumb in a network parameter to this class.

Reply:
While its updated in the next PR, your comment about the network stands. Its interesting though because it looks like the old networks are largely abandoned ~1 month after the new one is released and the update to this in the next PR is technically a new feature. Are you going to ensure new features work with old networks (In this case where it wouldn't break anything, just wouldn't surface the new errors)? That might be huge effort for little reward